platform/linuxbsd/os_linuxbsd.cpp

```diff
@@ -303,65 +303,150 @@ static String get_mountpoint(const String &p_path) {
+	result = execute("gvfs-trash", args, true, nullptr, nullptr, &err_code); // For older linux machines
+	if (result == OK && !err_code) {
+		return OK;
+	} else if (err_code == 2) {
+		return ERR_FILE_NOT_FOUND;
+	}
+
+	// If the commands `gio trash` or `gvfs-trash` don't exist on the system we do it manually
+	String trash_path = "";
 	String mnt = get_mountpoint(p_path);
 
-	// If there is a directory "[Mountpoint]/.Trash-[UID]/files", use it as the trash can.
+	// If there is a directory "[Mountpoint]/.Trash-[UID], use it as the trash can.
 	if (mnt != "") {
-		String path(mnt + "/.Trash-" + itos(getuid()) + "/files");
+		String path(mnt + "/.Trash-" + itos(getuid()));
 		struct stat s;
 		if (!stat(path.utf8().get_data(), &s)) {
-			trash_can = path;
+			trash_path = path;
 		}
 	}
 
-	// Otherwise, if ${XDG_DATA_HOME} is defined, use "${XDG_DATA_HOME}/Trash/files" as the trash can.
-	if (trash_can == "") {
+	// Otherwise, if ${XDG_DATA_HOME} is defined, use "${XDG_DATA_HOME}/Trash" as the trash can.
+	if (trash_path == "") {
 		char *dhome = getenv("XDG_DATA_HOME");
 		if (dhome) {
-			trash_can = String(dhome) + "/Trash/files";
+			trash_path = String(dhome) + "/Trash";
 		}
 	}
 
-	// Otherwise, if ${HOME} is defined, use "${HOME}/.local/share/Trash/files" as the trash can.
-	if (trash_can == "") {
+	// Otherwise, if ${HOME} is defined, use "${HOME}/.local/share/Trash" as the trash can.
+	if (trash_path == "") {
 		char *home = getenv("HOME");
 		if (home) {
-			trash_can = String(home) + "/.local/share/Trash/files";
+			trash_path = String(home) + "/.local/share/Trash";
 		}
 	}
 
 	// Issue an error if none of the previous locations is appropriate for the trash can.
-	if (trash_can == "") {
+	if (trash_path == "") {
 		ERR_PRINT("move_to_trash: Could not determine the trash can location");
 		return FAILED;
 	}
 
 	// Create needed directories for decided trash can location.
-	DirAccess *dir_access = DirAccess::create(DirAccess::ACCESS_FILESYSTEM);
-	Error err = dir_access->make_dir_recursive(trash_can);
-	memdelete(dir_access);
+	{
+		DirAccess *dir_access = DirAccess::create(DirAccess::ACCESS_FILESYSTEM);
+		Error err = dir_access->make_dir_recursive(trash_path);
+		Error err2 = dir_access->make_dir_recursive(trash_path + "/files");
+		Error err3 = dir_access->make_dir_recursive(trash_path + "/info");
+
+		// Issue an error if trash can is not created proprely.
+		if (err != OK || err2 != OK || err3 != OK) {
+			ERR_PRINT("move_to_trash: Could not create the trash can \"" + trash_path + "\"");
+			return err;
+		}
+		memdelete(dir_access);
+	}
 
-	// Issue an error if trash can is not created proprely.
-	if (err != OK) {
-		ERR_PRINT("move_to_trash: Could not create the trash can \"" + trash_can + "\"");
-		return err;
+	// The trash can is successfully created, now we check that we don't exceed our file name Length limit.
+	// If the file name it's too long trim it so we can add the identifyng number and ".trashinfo"
```

Comment:
Missing `.` at the end (coding style) + spelling: `is too long`, `identifying`

Reply:
Done.